zhaquirks/tuya/ts0001.py

```diff
@@ -0,0 +1,148 @@
+from typing import Any, Dict, Optional, Union
+
+from zigpy.profiles import zha
+import zigpy.types as t
+from zigpy.zcl import foundation
+from zigpy.zcl.clusters.general import Basic, Groups, Ota, Scenes, Time
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+from zhaquirks.tuya import TUYA_SEND_DATA
+from zhaquirks.tuya.mcu import (
+    DPToAttributeMapping,
+    EnchantedDevice,
+    TuyaMCUCluster,
+    TuyaOnOff,
+    TuyaEnchantableCluster,
+)
+
+
+class FingerBotMode(t.enum8):
+    CLICK = 0x00
+    SWITCH = 0x01
+    PROGRAM = 0x02
+
+
+class FingerBotReverse(t.enum8):
+    UP_ON = 0x00
+    UP_OFF = 0x01
+
+
+class TuyaFingerbotCluster(TuyaEnchantableCluster, TuyaMCUCluster):
+    attributes = TuyaMCUCluster.attributes.copy()
+    attributes.update({101: ("mode", FingerBotMode)})
+    attributes.update({102: ("down_movement", t.uint16_t)})
+    attributes.update({103: ("sustain_time", t.uint16_t)})
+    attributes.update({104: ("reverse", FingerBotReverse)})
+    attributes.update({105: ("battery", t.uint16_t)})
+    attributes.update({106: ("up_movement", t.uint16_t)})
+    attributes.update({107: ("touch_control", t.Bool)})
+
+    async def command(
+        self,
+        command_id: Union[foundation.GeneralCommand, int, t.uint8_t],
+        *args,
+        manufacturer: Optional[Union[int, t.uint16_t]] = None,
+        expect_reply: bool = True,
+        tsn: Optional[Union[int, t.uint8_t]] = None,
+        **kwargs: Any,
+    ):
+        """Override the default Cluster command."""
+
+        return await super().command(
+            TUYA_SEND_DATA,
+            *args,
+            manufacturer=foundation.ZCLHeader.NO_MANUFACTURER_ID,
```

Comment:
Can you confirm if the `NO_MANUFACTURER_ID` is needed too?

Reply:
Yup, and if it's really needed, I think you should also be able to have `TuyaFingerbotCluster` extend `NoManufacturerCluster`.

Reply:
It can be done, but it needs also overwritte the `command_id` to `TUYA_SEND_DATA` to make it work:
* https://github.com/zigpy/zha-device-handlers/issues/2493#issuecomment-1661338801